packages/backend/src/payment-method/ilp/connector/core/middleware/ilp-packet.ts

```diff
@@ -234,5 +235,34 @@ export function createIlpPacketMiddleware(
     ctx.assert(!ctx.body, 500, 'response body already set')
     ctx.assert(response.rawReply, 500, 'ilp reply not set')
     ctx.body = response.rawReply
+
+    if (ctx.services.telemetry && Number(prepare.amount)) {
+      ctx.services.telemetry?.incrementCounter('packet_count_prepare', 1, {
+        description: 'Count of incoming prepare packets'
+      })
+      if (response.fulfill) {
+        const { code, scale } = ctx.state.incomingAccount.asset
+        const value = BigInt(prepare.amount)
+        await ctx.services.telemetry.incrementCounterWithTransactionAmount(
+          'packet_amount_fulfill',
+          {
+            value,
+            assetCode: code,
+            assetScale: scale
+          },
+          {
+            description: 'Amount sent through the network',
+            valueType: ValueType.DOUBLE
+          }
+        )
+        ctx.services.telemetry?.incrementCounter('packet_count_fulfill', 1, {
```

Comment:
optional chaining not necessary since we checked ctx.services.telemetry
```suggestion
        ctx.services.telemetry.incrementCounter('packet_count_fulfill', 1, {
```